contracts/exchange/Exchange.sol

```diff
@@ -65,6 +65,46 @@ contract Exchange is IExchange, ContextUpgradeable, OwnableUpgradeable, ERC165St
         emit OrderPlaced(_msgSender(), id, _order);
     }
 
+    function fillOrder(
+        uint256 _orderId,
+        uint256 amountToFill
+    ) external override {
+        // Verify order exists
+        require(orderbook.verifyOrderExists(_orderId), "Non-existent order");
+
+        // Get order amounts and total payment
+        (uint256 orderAmount, uint volume) = orderbook.getOrderAmount(_orderId, amountToFill);
```

Comment:
Actually, you don't even need this. You can delete getOrderAmount() entirely.
Underneath line 79, where you get the order, you can just add the following:

uint256 orderAmount = order.= orders[_orderId].amountOrdered - orders[_orderId].amountFilled;
orderAmount = (amountToFill > orderAmount) ? orderAmount : amountToFill;
uint256 volume = orders[_orderId].price * orderAmount;